src/MensaWidget/MWFrame.class.st

```diff
@@ -11,43 +11,160 @@ Class {
 	#'squeak_changestamp' : 'Louis Christ 5/27/2024 19:43'
 }
 
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'tmc 6/6/2024 21:40'
+}
+MWFrame class >> cellSpacing [
+
+	^ 10
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'tmc 6/6/2024 21:36'
+}
+MWFrame class >> contentExtent [
+
+	^ 450 @ 550
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'tmc 6/6/2024 21:03'
+}
+MWFrame class >> contentMorphColor [
+
+	^ Color darkGray
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Oscar 6/5/2024 00:54'
+}
+MWFrame class >> contentMorphInnerBorder [
+
+	^ 5
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'tmc 6/6/2024 21:50'
+}
+MWFrame class >> contentPosition [
+
+	^ 5 @ 40
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Louis Christ 6/6/2024 19:50'
+}
+MWFrame class >> frameSize [
+
+	^ 460 @ 595
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Oscar 6/5/2024 01:01'
+}
+MWFrame class >> scrollBarThickness [
+
+	^ 20
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Oscar 6/5/2024 00:57'
+}
+MWFrame class >> spacingBetweenCanteenMorphs [
+
+	^ 5
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Oscar 6/5/2024 01:13'
+}
+MWFrame class >> windowHeaderSize [
+
+	^ 40 
+]
+
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Louis Christ 5/29/2024 18:43'
+	#'squeak_changestamp' : 'tmc 6/6/2024 21:37'
 }
-MWFrame >> addText [
+MWFrame >> buildFrameContent [
 
-	| textMorph |
+	| contentMorph |
 	
-	textMorph := TextMorph new.
-	textMorph contents: 'Are you a hungry?';
-		position:  135@100.
-	self addPaneMorph: textMorph.
+	contentMorph := ScrollPane new.
+	self fillContentMorph: contentMorph.
+	contentMorph scrollBarThickness: self class scrollBarThickness;
+		color: self class contentMorphColor;
+		position: self class contentPosition;
+		extent: self class contentExtent.
+	self addPaneMorph: contentMorph
 ]
 
 {
 	#category : #creation,
-	#'squeak_changestamp' : 'Louis Christ 5/27/2024 17:00'
+	#'squeak_changestamp' : 'tmc 6/6/2024 21:42'
 }
-MWFrame >> createFrame [
-		
-	self setLabel:'Mensa Widget';
-		openInWorld;
-		extent: 400@600.	
+MWFrame >> fillContentMorph: aContentMorph [
 	
+	| contentMorph |
 	
-	
-
+	contentMorph := AlignmentMorph newColumn.
+	contentMorph cellInset: self class cellSpacing;
```

Comment:
This line can already be in the cascade.

Reply:
No, because you would cascade on the class. but you want to cascade on the returned object of the first line.

Reply:
![image](https://github.com/hpi-swa-teaching/MensaWidget/assets/131597852/f4408841-661c-423a-853e-10570689db1f)